### PR TITLE
feat: Added example tf for customers

### PR DESCRIPTION
**Pull request by amanpruthi**
1. Standard TF deployment using k8s operator in GCP  - creating new VPC / VNet, new  GKE cluster, new MySQL cluster, and new Redis cache.
2. Custom TF deployment using k8s operator in  GCP  - in an existing VPC & pre-created subnets, connecting to an existing MySQL cluster, new GKE cluster, and new Redis cache.
3. Custom TF deployment using k8s operator in GCP- in an existing VPC & pre-created subnets, connecting to an existing MySQL cluster, using an existing GKE cluster, and an existing Redis cache.

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -11,46 +11,46 @@ output "bucket_queue_name" {
   description = "Pubsub queue created for google bucket file upload events."
 }
 output "cluster_ca_certificate" {
-  value       = module.app_gke.cluster_ca_certificate
+  value       = var.create_gke ? module.app_gke.0.cluster_ca_certificate : null
```
Comment (gls4) [inline outputs.tf:14]: Let's remove this conditional. The purpose of this repository is to setup kubernetes, and I don't foresee anyone capable of setting up kubernetes on their own needing our tf to setup anything else. We must assume we're always going to setup kubernetes.

**Review by gls4 — CHANGES_REQUESTED** [review]

File: outputs.tf (hunk 1 above)
Comment (amanpruthi) [reply outputs.tf:14]: @gls4  -- please check this link https://github.com/orgs/wandb/projects/17/views/1?pane=issue&itemId=60737094
It is a requirement that we should be able to install charts on the existing gke clusters(managed by some other terraform ) right ?

Comment (jsbroks) [reply outputs.tf:14]: @venky-wandb Can you explain why we're considering using Terraform to deploy into existing clusters? I'm hesitant about potential complexity. Do we have strong customer demand for this?

Comment (gls4) [reply outputs.tf:14]: @amanpruthi -- The purpose of this repo is to setup kubernetes and other resources. If kubernetes is already setup, this repo should not be used.

File: main.tf [hunk 2]
```diff
@@ -78,6 +78,7 @@ locals {
 }
 
 module "app_gke" {
+  count           = var.create_gke ? 1 : 0
```
Comment (jsbroks) [inline main.tf:81]: won't this recreate every cluster once we merge this?

**Pull request closed without merging**